invenio_i18n/selectors.py

```diff
@@ -70,8 +70,9 @@ def get_locale():
     if hasattr(current_app, 'login_manager') and \
             current_user.is_authenticated:
         language_user_key = current_app.config['I18N_USER_LANG_ATTR']
-        if getattr(current_user, language_user_key) in locales:
-            return getattr(current_user, language_user_key)
+        if hasattr(current_user, language_user_key):
+            if getattr(current_user, language_user_key) in locales:
```

Comment:
`if getattr(current_user, language_user_key, None) in locales:` would do it as well